python/feldera/_sql_table.py

```diff
@@ -0,0 +1,29 @@
+from typing import Optional
+from feldera.sql_schema import SQLSchema
+
+
+class SQLTable:
+    name: str
+    ddl: str
+    schema: Optional[SQLSchema]
+
+    def __init__(self, name: str, ddl: Optional[str] = None, schema: Optional[SQLSchema] = None):
+        if ddl is None and schema is None:
+            raise ValueError("Either ddl or schema must be provided")
+
+        self.name = name
+        self.ddl = ddl
+        self.schema = schema
+
+    def build_ddl(self):
+        """
+        Either returns the provided ddl or builds it from the schema
+        """
+
+        if self.schema is None:
+            return self.ddl
+
+        if self.ddl is not None:
```

Comment:
should this also be thrown in `__init__` rather than waiting for `build_ddl`

Reply:
Later we will likely have a method `infer_schema`, after which we will call `build_ddl`.

Reply:
Hmm, I expect that `infer_schema` will be connector-specific and will be part of whatever classes we build to represent connectors.